pkg/maps/policymap/policymap.go

```diff
@@ -441,6 +441,13 @@ func Create(path string) error {
 	return m.Create()
 }
 
+// Create unpinned creates a policy map without pinning it to the filesystem
+func CreateUnpinned(path string) (*PolicyMap, error) {
```

Comment:
Could the current uses of `policymap.Create()` be updated to be followed by a `Pin()` instead? There shouldn't be too many ways of creating a map.

Reply:
We could add an option to the current `Create` method, but that would involve updating many more callers. If the number of callers for `CreateUnpinned` grows in the codebase, we should consider consolidating the create logic in one place.

Reply:
Are there that many callers? 🤔 My point is CreateUnpinned() should be the default, with Pin() being an explicit action. Adding an option to Create() would be even worse. :)

Reply:
I don't have a strong feeling either way for this; the kernel separates create vs. pin and traditionally inside Cilium we haven't. Traditionally Cilium always pinned, then to handle the replacement of another map the pattern was to:
1. Keep the existing map + pin in place
2. Create a new map + new pin (`foo_pending` map name + pin on filesystem)
3. Once the new map is populated, delete the old map pin path and move the new one into the old one's place
4. If something fails, move the new pin to `foo_failed`. Then we can introspect the content on failure.

That pattern was quite useful for introspection early in the project, but I'll admit that I don't recall using the `foo_pending` / `foo_failed` map paths for years.

But that said, what I do feel more strongly about is if we're going to change the Create internal API for unrelated codepaths to change the meaning to avoid pinning and force every user to deal with the details of pinning, please submit that as a separate PR. This PR is complex enough as it is without including unrelated datapath pinning changes.